### ISSUE TITLE
Better HTTP/2 flow control server-side.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8681]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)| @Lukasa reported|
|-|-|
|Trac ID|trac#8681|
|Type|enhancement|
|Created|2016-07-26 08:26:45Z|
|Branch|https://github.com/twisted/twisted/tree/8681-lukasa-better-flow-control|

Right now the HTTP/2 server sends WINDOW_UPDATE frames for each DATA frame received (mostly). This is not the most efficient strategy for bulk transfer: it can lead to quite a lot of time spent writing these control frames. We should investigate having a better strategy.

<details><summary>Searchable metadata</summary>

```
trac-id__8681 8681
type__enhancement enhancement
reporter__Lukasa Lukasa
priority__normal normal
milestone__None None
branch__8681_lukasa_better_flow_control 8681-lukasa-better-flow-control
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1469521605314423 1469521605314423
changetime__1477646704622758 1477646704622758
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_1"></a>|@Lukasa commented|
|-|-|

In changeset 933fd40ab04f6ac0bd155e0318ce22e9e48e0cde
```
#!CommitTicketReference repository="" revision="933fd40ab04f6ac0bd155e0318ce22e9e48e0cde"
Topfile for #8681
```

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa commented|
|-|-|

As of hyper-h2 2.5.0 there is now builtin support for advanced flow control algorithms in the connection. This is a vast improvement over the current Twisted logic, and also reduces the amount of Twisted code required. A win win!

The patch is available in https://github.com/twisted/twisted/pull/571

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa commented|
|-|-|

In changeset 7cd2865a5efbc717d5b19cdfd76a7a2f2ac1ef7b
```
#!CommitTicketReference repository="" revision="7cd2865a5efbc717d5b19cdfd76a7a2f2ac1ef7b"
Topfile for #8681
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 7889f72e37495edde8cc3679403863b0cb293dcc
```
#!CommitTicketReference repository="" revision="7889f72e37495edde8cc3679403863b0cb293dcc"
Merge 8681-lukasa-better-flow-control: Better HTTP/2 flow control.

Author: lukasa
Reviewer: hawkowl
Fixes: #8681
```
